v2/yarpcgrpc/integration_test.go

```diff
@@ -21,521 +21,391 @@
 package grpc
 
 import (
-	"bytes"
 	"context"
 	"crypto/ecdsa"
 	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/tls"
 	"crypto/x509"
 	"crypto/x509/pkix"
+	"errors"
 	"fmt"
-	"math"
 	"math/big"
 	"net"
+	"net/url"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
-	"github.com/gogo/protobuf/proto"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/multierr"
-	"go.uber.org/yarpc/api/transport"
-	"go.uber.org/yarpc/encoding/protobuf"
-	"go.uber.org/yarpc/internal/clientconfig"
-	"go.uber.org/yarpc/internal/examples/protobuf/example"
-	"go.uber.org/yarpc/internal/examples/protobuf/examplepb"
-	"go.uber.org/yarpc/internal/grpcctx"
-	"go.uber.org/yarpc/internal/testtime"
-	intyarpcerrors "go.uber.org/yarpc/internal/yarpcerrors"
-	"go.uber.org/yarpc/peer"
-	"go.uber.org/yarpc/peer/hostport"
-	"go.uber.org/yarpc/pkg/procedure"
-	"go.uber.org/yarpc/yarpcerrors"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
+	"go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpcerror"
+	"go.uber.org/yarpc/v2/yarpcjson"
+	"go.uber.org/yarpc/v2/yarpctest"
 	"google.golang.org/grpc/credentials"
-	"google.golang.org/grpc/status"
 )
 
-func TestYARPCBasic(t *testing.T) {
-	t.Parallel()
-	te := testEnvOptions{}
-	te.do(t, func(t *testing.T, e *testEnv) {
-		_, err := e.GetValueYARPC(context.Background(), "foo")
-		assert.Equal(t, yarpcerrors.Newf(yarpcerrors.CodeNotFound, "foo"), err)
-		assert.NoError(t, e.SetValueYARPC(context.Background(), "foo", "bar"))
-		value, err := e.GetValueYARPC(context.Background(), "foo")
-		assert.NoError(t, err)
-		assert.Equal(t, "bar", value)
-	})
-}
-
-func TestGRPCBasic(t *testing.T) {
+func TestYARPCMaxMsgSize(t *testing.T) {
 	t.Parallel()
-	te := testEnvOptions{}
-	te.do(t, func(t *testing.T, e *testEnv) {
-		_, err := e.GetValueGRPC(context.Background(), "foo")
-		assert.Equal(t, status.Error(codes.NotFound, "foo"), err)
-		assert.NoError(t, e.SetValueGRPC(context.Background(), "foo", "bar"))
-		value, err := e.GetValueGRPC(context.Background(), "foo")
-		assert.NoError(t, err)
-		assert.Equal(t, "bar", value)
-	})
-}
 
-func TestTLSWithYARPCAndGRPC(t *testing.T) {
 	tests := []struct {
-		clientValidity      time.Duration
-		serverValidity      time.Duration
-		expectedErrContains string
-		name                string
+		name string
```

Comment:
nit: traditionally I've seen `desc` used here instead of `name`.

Reply:
not a fan